src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1554,11 +1558,60 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
 
 /*
  * SetWorkerColumn function sets the column with the specified index
+ * on the worker in pg_dist_node, by calling SetWorkerColumnLocalOnly.
+ * It also sends the same command for node update to other metadata nodes.
+ * Returns the new worker node after the modification.
+ */
+WorkerNode *
+SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
+{
+	workerNode = SetWorkerColumnLocalOnly(workerNode, columnIndex, value);
+
+	char *metadataSyncCommand = GetMetadataSyncCommand(workerNode, columnIndex, value);
+
+	SendCommandToWorkersWithMetadata(metadataSyncCommand);
+
+	return workerNode;
+}
+
+
+/*
+ * SetWorkerColumnOptional function sets the column with the specified index
+ * on the worker in pg_dist_node, by calling SetWorkerColumnLocalOnly.
+ * It also sends the same command optionally for node update to other metadata nodes.
+ * Returns the new worker node after the modification.
+ */
+WorkerNode *
+SetWorkerColumnOptional(WorkerNode *workerNode, int columnIndex, Datum value)
+{
```

Comment:
Hmm, I think we should not call `SetWorkerColumnLocalOnly` immediately. If `SendOptionalCommandListToWorkerInCoordinatedTransaction` fails, I guess we should not be calling `SetWorkerColumnLocalOnly`. Because, the metadata would not be properly synced to the `workerNode`.

I know that the possibility of this pretty low, a failure should happen right after `SyncMetadataSnapshotToNode` and right before we send `metadataSyncCommand` on the `workerNode`. And, sometimes databases fail :)

What do you think?